### PR TITLE
DllPlugin & DllReferencePlugin docs

### PR DESCRIPTION
**Pull request by aretecode**
Feedback appreciated

### CODE REVIEW

**Comment by simon04** [comment]
:rocket: Great to see progress on this big black hole in the documentation of webpack 2 (#944). Over the weekend, I was also working on finally getting the `DllPlugin` documented.

Firstly, don't worry about the failing Travis build (this is due to an edit link of the new markdown file not yet existing).

I see that this documentation is largely based on https://webpack.github.io/docs/list-of-plugins.html#dllplugin which seems a good start. Since the successful usage involves a correct configuration of `DllPlugin`, `DllReferencePlugin` as well as the use in the JavaScript code and HTML file, a few examples (for scoped, mapped mode, various `output.library` values) covering all parts would be of greatest help for the reader. I could think of the following structure of the documentation:
* introduction (as it is)
* configuration options for `DllPlugin`, `DllReferencePlugin` (as it is) w/o an example
* a few examples showing two complete webpack configs plus minimal JavaScript files plus the inclusion in HTML file

The question is how much time you'd be willing to invest for this documentation/PR? So whether you'd like to get this merged quickly or you're up to working and polishing this docs for some time.

**Comment by aretecode** [comment]
@simon04 

yes, it is based on the old documentation, with a fair bit of updating.

thanks for the info about travis.

where are the docs you made for the dll plugin? we could merge them :-)

I had planned to add some examples to webpack/examples here https://github.com/webpack/webpack.js.org/pull/1121/files#diff-0dcb8884c3201edf1c374f1ea874abe3R76

I could port the two examples I have for my `d-l-l` lib https://github.com/fliphub/d-l-l/tree/master/examples ?

**Comment by simon04** [comment]
I stopped to write the docs right after I saw your PR. So there's not much to show. I summarised my ideas in my previous comment.

It would be good to have one simple+minimal example (I like this one https://gist.github.com/Eoksni/83d1f1559e0ec00d0e89c33a6d763049) which shows the basic usage. Altered examples should show the other modes (scoped, mapped mode), and how `sourceType='commonjs2'` is to be used.

Update this PR with some good examples and I'll review.

**Comment by aretecode** [comment]
https://github.com/webpack/webpack/pull/4745
https://github.com/webpack/webpack/pull/4746

**Comment by aretecode** [comment]
@simon04 [updated here](https://github.com/webpack/webpack.js.org/pull/1121/commits/440b6a77a1019694cbd048284d940e6b042201e5) along with [adding & updating examples to webpack/examples](https://github.com/webpack/webpack.js.org/pull/1121#issuecomment-295229328)

**Comment by simon04** [comment]
Aside from links to webpack examples added, I don't see any changes in the page structure (see https://github.com/webpack/webpack.js.org/pull/1121#issuecomment-294779817) nor a change in the documentation of scoped mode, mapped mode and how to use `sourceType='commonjs2'`.

File: content/plugins/dll-plugin.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+contributors:
+  - aretecode
+  - sokra
+  - opiepj
+  - simon04
+---
+
+The `DllPlugin` and `DllReferencePlugin` provide means to split bundles in a way that can drastically improve build time performance.
+
+[DllPlugin](#DllPlugin) is used in a separate webpack config exclusively to create a dll-only-bundle. It creates a `manifest.json` file, which used by the [DllReferencePlugin](#DllReferencePlugin) to map dependencies.
```
Comment (smashercosmo) [inline content/plugins/dll-plugin.md]: which used -> which is used

Comment (aretecode) [inline content/plugins/dll-plugin.md]: thanks

**Comment by aretecode** [comment]
@simon04 any thoughts? has been updated

**Comment by skipjack** [comment]
@aight8 any thoughts on this? I know you brought it up in #944 which can probably be closed once this is merged..

**Comment by simon04** [comment]
Since this PR references many not-yet-merged webpack examples, the Travis CI fails (HTTP 404). Feel free to drop those examples for now and submit another PR later.

I won't review unless the CI succeeds. Also please make sure to read and follow https://webpack.js.org/writers-guide/#typesetting.

**Comment by skipjack** [comment]
@aretecode please re-open once you've rebased and dealt with the conflicts (or just open a new pr).

**Pull request closed without merging**